### ISSUE TITLE
Enhancement: Add a JSON context for Convert JSON methods

### ISSUE THREAD

**Issue by robbieaverill** [silverstripe/silverstripe-framework#6550]
It would be nice to be able to provide a context for JSON [options](http://php.net/manual/en/function.json-encode.php) such as `JSON_PRETTY_PRINT`, `JSON_UNESCAPED_SLASHES` etc.

E.g.
```php
echo Convert::raw2json($myVar, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT);
```

**Comment by robbieaverill** [comment]
I'd originally thought that this would apply to `json_encode` and `json_decode`, however the decode method only supports two options and they are reasonably unlikely to be used in SilverStripe, so I'll just implement this in the encode method.

Also I'd imagined this being encapsulated in a `JsonContext` class which would return the bitmask on invocation, but it seems like a bunch of work to basically just wrap up basic PHP assignments and arithmetic, so I flagged it.

**Comment by dhensby** [comment]
fixed in #6551
